staging/src/k8s.io/client-go/tools/leaderelection/resourcelock/interface.go

```diff
@@ -90,14 +90,20 @@ type Interface interface {
 	Describe() string
 }
 
-// Manufacture will create a lock of a given type according to the input parameters
+// New will create a lock of a given type according to the input parameters
 func New(lockType string, ns string, name string, coreClient corev1.CoreV1Interface, coordinationClient coordinationv1.CoordinationV1Interface, rlc ResourceLockConfig) (Interface, error) {
+	return NewWithLabels(lockType, ns, name, map[string]string{}, coreClient, coordinationClient, rlc)
+}
+
+// NewWithLabels will create a lock with labels of a given type according to the input parameters
+func NewWithLabels(lockType string, ns string, name string, labels map[string]string, coreClient corev1.CoreV1Interface, coordinationClient coordinationv1.CoordinationV1Interface, rlc ResourceLockConfig) (Interface, error) {
```

Comment:
if a component starts using this on an upgrade boundary, will the old component (using New) and new component (using NewWithLabels) dual with each other over the label content of the lease (will the old version of the component drop the labels on update)?